backend/data-preparation/data_collector.py

```diff
@@ -0,0 +1,159 @@
+    
+    try:
+        url_parts = url.split('?')[0].split('#')[0]
+        path_segments = [segment for segment in url_parts.split('/') if segment]
+        
+        if "github.com" in path_segments:
+            github_index = path_segments.index("github.com")
+            if len(path_segments) > github_index + 2:
+                owner = path_segments[github_index + 1]
+                repo = path_segments[github_index + 2]
+                
+                if repo.endswith(".git"):
+                    repo = repo[:-4]
+                return owner, repo
+            
+        logger.warning(f"Could not extract owner and repo from GitHub URL: {url}")
+        return None
+    except Exception as e:
+        logger.error(f"Error parsing GitHub URL {url}: {e}")
+        return None
+    
+parsed_repo = _extract_owner_repo_from_url(TARGET_GITHUB_REPO_URL)
+NIGERIAN_LAWS_GITHUB_REPOS_FOR_LOADER: List[tuple[str, str]] = []
+if parsed_repo:
+    NIGERIAN_LAWS_GITHUB_REPOS_FOR_LOADER.append(parsed_repo)
+else:
+    logger.critical("Failed to parse the target GitHub URL")
+    
+class DataCollector:
+    
+    def __init__(self):
+        
+        self.db_manager = NigerianLawsDatabase(MONGO_URI, MONGO_DB_NAME)
+        self.collection = self.db_manager.get_collection(MONGO_COLLECTION_NAME)
+        
+        logger.info("DataCollector initialized successfully.")
+    
+    def save_document(self, data: Dict) -> bool:
+        
+        unique_identifier = data.get('url')
+        
+        try:
+            result = self.collection.update_one(
+                {'url': unique_identifier},
+                {'$set': data},
+                upsert=True
+            )
+            
+            if result.upserted_id:
+                logger.info(f"Inserted new document: {unique_identifier} (ID: {result.upserted_id})")
+            elif result.modified_count > 0:
+                logger.info(f"Updated existing document: {unique_identifier}")
+            else:
+                logger.debug(f"Document: {unique_identifier} already exists and no changes detected.")
+            return True
+        
+        except Exception as e:
+            logger.error(f"Error saving document to MongoDB for {unique_identifier}: {e}")
+            return False
+        
+    def collect_github_repo_data(self, repo_url: str, branch: str = "master"):
+        
+        if not repo_url:
+            logger.info("No GitHub repository URL provided")
+            return
+        
+        try:
+            clean_url = repo_url.split('?')[0]
+            path_parts = clean_url.strip('/').split('/')
+            owner = path_parts[-2]
+            repo_name = path_parts[-1]
+            
+            if not owner or not repo_name:
+                raise ValueError("Could not parse owner and repo_name from provided URL.")
+            
+            repo_full_url = f"https://github.com/{owner}/{repo_name}"
```

Comment:
we know the url, why not just reference it straight? why build it? 
```suggestion
            repo_full_url = f"https://github.com/mykeels/nigerian-laws"
```